Wordle.Api/Wordle.Api/Services/LeaderboardService.cs

```diff
@@ -0,0 +1,50 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.EntityFrameworkCore;
+using Wordle.Api.Data;
+using Wordle.Api.Dtos;
+
+namespace Wordle.Api.Services
+{
+    public class LeaderboardService
+    {
+        private readonly AppDbContext _db;
+
+        public LeaderboardService(AppDbContext db)
+        {
+            _db = db;
+        }
+
+        public async Task<List<Player>> GetTopPlayers(int count)
+        {
+            return await _db.Players.OrderByDescending(p => p.AverageAttempts).Take(count).ToListAsync();
```

Comment:
If score is AverageAttempts, then OrderByDescending is the wrong direction. You may also want to require the player to have a min num of games as a threshold before showing them on the leaderboard. If you do this, be sure to filter first before sorting.

Reply:
ive updateing my score and sorting methods, thanks for the tip